### PR TITLE
[TTAHUB-17] Approvers can edit ARs in submitted state

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Approvers can edit activity reports that have been submitted IF no other approver has marked it as "approved" or "needs action."

Will still need to check with the larger group as to what we want the timing of this merge to be, and if we need some kind of simultaneous editing solution in place before we put this forward.

## How to test
Create an activity report with one or two approvers, submitting it. 
Test the following:
- An approver can make changes if the report has been submitted
- An approver can't make changes to reports in any other status
- An approver can't make changes once another approver has reviewed (either "needs action" or "approved"
- The rest of the AR flow seems to work as normal

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-17

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/index.js
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/index.js:214]: Is the check for report status needed? I see a similar check for 'approverHasMarkedReport'.

Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/index.js:214]: ```approverHasMarkedReport``` is checking _approval_ status, the other checks are for report status. 

Collaborators & authors can edit when the report status is draft or needs action, approvers can only edit in the submitted state. Not sure if that answers your question or if I'm missing a way to clean it up a little further

Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/index.js:214]: Yea far as I know there are two statuses Submission and Calculated. Submission status is more or less static. Calculated status reflects all the approver statuses. Yea so I think you are correct.

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good. I was able to create a report and edit it while in the submitted state as a approver.

One thing I noticed is that if the creator is also an approver there seems to be no way to get it back into a draft state. Not sure if this is a real case we need to worry about.

**Comment by jasalisbury** [comment]
I was able to "Approve" a report with pages that are not "complete". We need to disable the submit button and show the same warning to approvers that we show submitters when the report is incomplete.

**Comment by thewatermethod** [comment]
> I was able to "Approve" a report with pages that are not "complete". We need to disable the submit button and show the same warning to approvers that we show submitters when the report is incomplete.

I think I solved this, if the path that you took to make that happen was "incompleting" a report after it was submitted as complete. Let me know if there is another way to get into that situation that I'm not accounting for

**Review by jasalisbury — APPROVED** [review]
:ship:

**Comment by thewatermethod** [comment]
Spaced on talking this through with the group, didn't know if we needed to wait for some kind of simultaneous editing alert before merging this in. Will add it to standup agenda for Monday

**Pull request merged**
